lib/reader.ts

```diff
@@ -377,13 +394,13 @@ class ParquetEnvelopeReader {
     return new ParquetEnvelopeReader(readFn, closeFn, fileStat, options);
   }
 
-  static async openUrl(params, options) {
+  static async openUrl(params: Parameter, options: BufferReaderOptions) {
      if (typeof params === 'string')
       params = {url: params};
     if (!params.url)
       throw new Error('URL missing');
 
-    let base = params.url.split('/');
+    let base: string | string[] = params.url.split('/');
     base = base.slice(0, base.length-1).join('/')+'/';
```

Comment:
One way to get rid of the ambiguity:
```suggestion
    const baseArr = params.url.split('/');
    const base = baseArr.slice(0, baseArr.length-1).join('/')+'/';
```